### PR TITLE
[IMP] add branch 11.0

### PR DESCRIPTION
**Pull request by Danisan**
[FIX] modifications to work correctly in 11.0
[IMP] add different currency check
[FIX] does not considerate the date in the due amounts

### CODE REVIEW

**Comment by JayVora-SerpentCS** [comment]
@Swapnesh-SerpentCS review plz.

File: partner_credit_limit/models/partner.py [hunk 1]
```diff
@@ -1,9 +1,14 @@
+# -*- coding: utf-8 -*-
 # See LICENSE file for full copyright and licensing details.
-
-from odoo import fields, models
+from odoo import _, fields, models
```
Comment (Swapnesh-SerpentCS) [inline partner_credit_limit/models/partner.py:3]:
```suggestion
from odoo import fields, models, _
```

File: partner_credit_limit/models/partner.py [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
 
 
 class ResPartner(models.Model):
+    _name = 'res.partner'
```
Comment (Swapnesh-SerpentCS) [inline partner_credit_limit/models/partner.py:7]: Is it really needed?

File: partner_credit_limit/models/partner.py [hunk 3]
```diff
@@ -1,9 +1,14 @@
...
     _inherit = 'res.partner'
 
-    over_credit = fields.Boolean('Allow Over Credit?')
+    credit_limit = fields.Float('Credit Limit',
```
Comment (Swapnesh-SerpentCS) [inline partner_credit_limit/models/partner.py:10]: Better to add `copy=False`
Do you mean `local amount` as `Local currency`?

File: partner_credit_limit/models/sale.py [hunk 4]
```diff
@@ -1,44 +1,55 @@
...
+import logging
 from odoo import api, models, _
 from odoo.exceptions import UserError
-from odoo.tools import DEFAULT_SERVER_DATE_FORMAT as DF
+
+_logger = logging.getLogger(__name__)
 
 
 class SaleOrder(models.Model):
+    _name = 'sale.order'
```
Comment (Swapnesh-SerpentCS) [inline partner_credit_limit/models/sale.py:11]: Same here

File: partner_credit_limit/models/sale.py [hunk 5]
```diff
@@ -1,44 +1,55 @@
...
     _inherit = "sale.order"
 
-    @api.multi
     def check_limit(self):
         self.ensure_one()
-        partner = self.partner_id
-        moveline_obj = self.env['account.move.line']
-        movelines = moveline_obj.search(
+        if self.partner_id.commercial_partner_id:
```
Comment (Swapnesh-SerpentCS) [inline partner_credit_limit/models/sale.py:16]: `commercial_partner_id` will be always there (see `_compute_commercial_partner`)

File: partner_credit_limit/models/sale.py [hunk 6]
```diff
@@ -1,44 +1,55 @@
...
+            partner = self.partner_id.commercial_partner_id
+        else:
+            partner = self.partner_id
+        move_lines = self.env['account.move.line'].search(
             [('partner_id', '=', partner.id),
-             ('account_id.user_type_id.name', 'in', ['Receivable', 'Payable']),
-             ('full_reconcile_id', '=', False)]
+             ('account_id.user_type_id.id', 'in', [
+                 self.env.ref('account.data_account_type_receivable').id,
```
Comment (Swapnesh-SerpentCS) [inline partner_credit_limit/models/sale.py:23]: Why use hardcoded Ids for the account ?
We should use account type only.

File: partner_credit_limit/models/sale.py [hunk 7]
```diff
@@ -1,44 +1,55 @@
...
+                 self.env.ref('account.data_account_type_payable').id]),
+             ('full_reconcile_id', '=', False),
+             ('company_id', '=', self.env.user.company_id.id)]
```
Comment (Swapnesh-SerpentCS) [inline partner_credit_limit/models/sale.py:26]: Do we really need User's company here? (or atleast SO's company)

File: partner_credit_limit/models/sale.py [hunk 8]
```diff
@@ -1,44 +1,55 @@
...
         )
+
         debit, credit = 0.0, 0.0
-        today_dt = datetime.strftime(datetime.now().date(), DF)
-        for line in movelines:
-            if line.date_maturity < today_dt:
-                credit += line.debit
-                debit += line.credit
+        for line in move_lines:
```
Comment (Swapnesh-SerpentCS) [inline partner_credit_limit/models/sale.py:30]: Would be better to use `search_read` as we are doing this  for `credit`/`debit` only

File: partner_credit_limit/models/sale.py [hunk 9]
```diff
@@ -1,44 +1,55 @@
...
+            credit += line.debit
+            debit += line.credit
 
-        if (credit - debit + self.amount_total) > partner.credit_limit:
+        if self.currency_id == self.env.user.company_id.currency_id:
+            amount_total = self.amount_total
+        else:
+            amount_total = self.amount_total / self.currency_id.rate
+        _logger.info(
```
Comment (Swapnesh-SerpentCS) [inline partner_credit_limit/models/sale.py:38]: To be removed

File: partner_credit_limit/models/sale.py [hunk 10]
```diff
@@ -1,44 +1,55 @@
...
+                      '%s.\nCheck Partner Accounts or Credit ' \
+                      'Limits !' % (credit - debit)
+                raise UserError(_('Credit Over Limits:\n' + msg))
+            partner.write(
+                {'credit_limit': credit - debit + amount_total, 'over_credit': False, })
         return True
 
     @api.multi
     def action_confirm(self):
+        for record in self:
```
Comment (Swapnesh-SerpentCS) [inline partner_credit_limit/models/sale.py:52]: `record` --> `order`

File: partner_credit_limit/security/credit_permissions.xml [hunk 11]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data noupdate="0">
+        <record id="credit_limit_permissions" model="res.groups">
+            <field name="name">Edit credit limit and over-credit value for partners</field>
+        </record>
+    </data>
+</odoo>
```
Comment (Swapnesh-SerpentCS) [inline partner_credit_limit/security/credit_permissions.xml:8]: Add new line in the end

File: partner_credit_limit/security/credit_permissions.xml [hunk 12]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data noupdate="0">
+        <record id="credit_limit_permissions" model="res.groups">
+            <field name="name">Edit credit limit and over-credit value for partners</field>
```
Comment (Swapnesh-SerpentCS) [inline partner_credit_limit/security/credit_permissions.xml:5]:
```suggestion
            <field name="name">Manage Credit Limits for Partners</field>
```

**Pull request closed without merging**